tests/server-tests/api_test.go

```diff
@@ -101,20 +96,80 @@ func testMethod(t *testing.T, testInput TestInput) error {
 			testInput.method, testInput.path, testInput.expectedStatusCode, resp.Status)
 	}
 
+	if testInput.method == "GET" { // Are there other methods which get content. We could genericize
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			// handle error
+			t.Errorf("testMethod: Error on ioutil.ReadAll: %v", err)
+		}
+
+		if testInput.expectedStatusCode != 404 &&
+			len(testInput.content) > 0 &&
+			string(body) != testInput.content {
+			t.Errorf("testMethod: Error, expected content %v, got %v", testInput.content, resp.Body)
+		}
+	}
+
 	testOutput.status = resp.Status
 	testOutput.statusCode = resp.StatusCode
 
 	return err
 }
 
+func TestPaths(t *testing.T) {
+	type PathResult struct {
+		path       string
+		statusCode int
+	}
+
+	var testInput TestInput
+	var testOutput TestOutput
+	var paths []PathResult
+
+	fmt.Println("TestPaths")
+
+	testInput.client = getClient()
+
+	paths = make([]PathResult, 7)
+	paths[0].path = getServerPath()
+	paths[0].statusCode = 200
+	paths[1].path = paths[0].path + "sites/"
+	paths[1].statusCode = 405
+	paths[2].path = paths[1].path + getSiteId() + "/"
+	paths[2].statusCode = 404
+	paths[3].path = paths[2].path + "/environments/"
+	paths[3].statusCode = 405
+	paths[4].path = paths[3].path + "self/"
+	paths[4].statusCode = 404
+	paths[5].path = paths[3].path + "dev/"
+	paths[5].statusCode = 404
+	paths[6].path = paths[5].path + "files/"
+	paths[6].statusCode = 404
+
+	testInput.content = "" // content is ignored
+	testInput.method = "GET"
+
+	// for _, method := range []string{"HEAL", "TEST", "LOCK", "PROPPATCH", "OPTIONS"} {
+	// Temporarily remove HEAL, it takes a long time
+	for _, entry := range paths {
+		testInput.path = entry.path
+		testInput.expectedStatusCode = entry.statusCode
+		err := testMethod(t, testInput)
+		if err != nil {
+			// handle error
```

Comment:
I'm not sure if ignoring an error here is going to cause pain in debugging later.